src/common/asb/Asb.c

```diff
@@ -1056,6 +1061,36 @@ void AsbInitialize(OsConfigLogHandle log)
     OsConfigLogInfo(log, "%s initialized", g_asbName);
 }
 
+static char* GetModuleDirectory()
+{
+    Dl_info dl_info;
+    char* directory = NULL;
+    char* lastSlash = NULL;
+    size_t directoryLength = 0;
+
+    // Get information about the current function's address
+    if (dladdr((void*)GetModuleDirectory, &dl_info) != 0)
+    {
+        if (dl_info.dli_fname != NULL)
+        {
+            // Find the last slash to get the directory path
+            lastSlash = strrchr(dl_info.dli_fname, '/');
+            if (lastSlash != NULL)
+            {
+                directoryLength = lastSlash - dl_info.dli_fname;
+                directory = malloc(directoryLength + 1);
+                if (directory != NULL)
+                {
+                    strncpy(directory, dl_info.dli_fname, directoryLength);
+                    directory[directoryLength] = '\0';
+                }
+            }
+        }
+    }
+
+    return directory; // Caller must free this memory
+}
```

Comment:
Move this to the new telemetry static lib -- it will be needed by the other consumers as well.

Split the lib into its own PR. 

Have a separate PR for adding to ASB, SSH, etc.